### ISSUE TITLE
Fails Every download

### ISSUE THREAD

**Issue by Archie-2021** [ejv2/podbit#11]
Hello and thanks for your effort for this program

I installed it to use yt-dlp with newsboat. unfortunatly every single download fails even before starting (even for normal files without a + before their url). The things that download easily with podboat. I also get an error for every single file in my home directory like this when I run podbit:

```
Error: Invalid media file "~/bashrc" in cache! Ignoring...
```

Any idea what is wrong here?

**Comment by ejv2** [comment]
Hi. Sorry you're having issues. I believe this may be caused by attempting to use a directory which contains many other files as the download destination. Podbit kind of assumes that it has exclusive use of the destination directory (hence attempting to load your bashrc as a media file and failing). Can you try setting another empty directory as your download directory in your newsboat config file? The instructions for doing this are located in the podboat man page.

**Comment by Archie-2021** [comment]
> Hi. Sorry you're having issues. I believe this may be caused by attempting to use a directory which contains many other files as the download destination. Podbit kind of assumes that it has exclusive use of the destination directory (hence attempting to load your bashrc as a media file and failing). Can you try setting another empty directory as your download directory in your newsboat config file? The instructions for doing this are located in the podboat man page.

I don't think that being the issue as I have a directory set for podboat and it is working fine. (it is not set to use my home directory). And to clarify, it is not only want to read bashrc. it is sending that message for every single file in my home directory. like every random file that is inside my ~.

The only like of related config I have related to podboat is this:

`download-path "<some path>/audio/podcasts/%n"

Thanks

**Comment by ejv2** [comment]
> And to clarify, it is not only want to read bashrc. it is sending that message for every single file in my home directory. like every random file that is inside my ~.

Yes that's what I meant. It is trying to use this as a cache and is then attempting to load all files as cached media files, which obviously fails.

> The only like of related config I have related to podboat is this:
> 
> ```
> download-path "<some path>/audio/podcasts/%n"
> player "mpv"
> ```
The formatting in the download path is likely the problem. I haven't gotten around to implementing that in podbit. I'll add it to the todo list. In the meantime, try using a path which doesn't contain the format string options  (i.e: "%n").

The problem is that if the provided directory doesn't exist, podbit just uses your home directory instead.

**Comment by Archie-2021** [comment]
> > And to clarify, it is not only want to read bashrc. it is sending that message for every single file in my home directory. like every random file that is inside my ~.
> 
> Yes that's what I meant. It is trying to use this as a cache and is then attempting to load all files as cached media files, which obviously fails.
> 
> > The only like of related config I have related to podboat is this:
> > ```
> > download-path "<some path>/audio/podcasts/%n"
> > player "mpv"
> > ```
> 
> The formatting in the download path is likely the problem. I haven't gotten around to implementing that in podbit. I'll add it to the todo list. In the meantime, try using a path which doesn't contain the format string options (i.e: "%n").
> 
> The problem is that if the provided directory doesn't exist, podbit just uses your home directory instead.

I see, thanks for you help. tbh after I read your readme I learnt a lot about how newsboat works with queue file. Looking forward to see that update.

Thanks

**Comment by plum** [comment]
This  could be perfect. I also get the 'Failed (IO Error)' warning.

Newsboat Config:
### podboat integration ###
max-downloads 3
player mpv
delete-played-files yes
download-path /Home/users/chuee/Podcasts/podbit

----
Archlinux, wayland, pipewire.


Cheers.
John

**Comment by ejv2** [comment]
Hi John. Would you mind posting the contents of your queue file for me? That error should only occur if podbit fails to create the destination file for a queue entry.

I think that this could occur if your download directory doesn't exist yet, so you could try that as well.

Thanks!

**Comment by plum** [comment]
Hello @ejv2  Ethan, 
Thanks for kind reply. Your're suggestion tipped me off , and I sorted it all out. 
I had an old queue with itemswhose urls no longer exist; it sbeen a whle since i used podboat.
Now its working very nicely, wiht some frewsh 'enqued items' thank you.

Incidentally, May I ask, is the 'enque' key command available only  in an open item in  Newsboat the only way to add to newsboats download queue? I suppose it is.
John

**Comment by ejv2** [comment]
Hi John. Glad you got it sorted out. I've wanted to be able to add items to the queue from the RSS listing for a while but I don't think it's possible with stock newsboat. I had an idea of making a macro which opens the item, enqueues it and then closes it with one button press but I never got around to making it. Thanks again for the help!

**Comment by ejv2** [comment]
I'm going to close this issue to keep my list clear. If there are any further issues, please feel free to re-open or create a new issue.
